ci/build-project.sh

```diff
@@ -0,0 +1,50 @@
+#!/usr/bin/env bash
+
+set -e -u -o pipefail
+if [[ -n "${DEBUG-}" ]]; then
+    set -x
+fi
+
+pushd "$(dirname "$0")"
+    UNREAL_PATH="${1?Please enter the path to the Unreal Engine.}"
+    TEST_REPO_BRANCH="${2?Please enter the branch that you want to test.}"
+    TEST_REPO_URL="${3?Please enter the URL for the git repo you want to test.}"
+    TEST_REPO_UPROJECT_PATH="${4?Please enter the path to the uproject inside the test repo.}"
+    TEST_REPO_PATH="${5?Please enter the path where the test repo should be cloned to.}"
+    GDK_HOME="${6?Please enter the path to the GDK for Unreal repo.}"
+    BUILD_PLATFORM="${7?Please enter the build platform for your Unreal build.}"
+    BUILD_STATE="${8?Please enter the build state for your Unreal build.}"
+    BUILD_TARGET="${9?Please enter the build target for your Unreal build.}"
+
+    # Clone the testing project
+    echo "Cloning the testing project from ${TEST_REPO_URL}"
+    git clone \
+        --branch "${TEST_REPO_BRANCH}" \
+        "${TEST_REPO_URL}" \
+        "${TEST_REPO_PATH}" \
+        --single-branch
```

Comment:
`--depth 1` ? This also implies `--single-branch` according to the docs. Although in the future, I think we would like pinning to commits rather than branches (doesn't have to be now).

Reply:
> I think we would like pinning to commits rather than branches

+100 on making builds more reproducible

Reply:
Haha, but we had commit pinning for the engine, and everyone disliked that from a development perspective, so we would have to make some kind of hybrid.

Reply:
@petemounce  suggested `--single-branch`. I don't have a strong opinion

 I would also be for commit pinning. We would just need to figure out a sane solution across the 3 repos